ahcore/models/jit_model.py

```diff
@@ -1,24 +1,231 @@
+        self._fm_type = model_type
+        self._fm_embed_type = FMEmbedType.for_type(self._fm_type)
+        if self._output_mode == FMOutputMode.EMBED_CLS_ONLY:
+            self._forward_function = self._cls_only
+        elif self._output_mode == FMOutputMode.EMBED_PATCH_ONLY:
+            self._forward_function = self._patch_only
+        elif self._output_mode == FMOutputMode.EMBED_CONCAT:
+            self._forward_function = self._concat_embedding
+        elif self._output_mode == FMOutputMode.GENERAL_EMBEDDING:
+            self._forward_function = self._general_embedding
+        else:
+            raise NotImplementedError(f"Output mode {self._output_mode} is not supported for {self.__class__.__name__}")
+
+    @classmethod
+    def from_jit_path(cls, jit_path: Path, model_type: AllModelTypes, output_mode: AllOutputModes) -> Any:
+        model = load(jit_path)  # type: ignore
+        return cls(model, model_type, output_mode)
+
+    def _cls_only(self, x: torch.Tensor) -> Any:
+        """
+        Extract the class token embedding from the foundation model.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            The input tensor to the model.
+
+        Returns
+        -------
+        The class token embedding.
+        """
+        return self._model(x)[FMEmbedType.CLS_TOKEN.value]
+
+    def _patch_only(self, x: torch.Tensor) -> Any:
+        """
+        Extract the patch token embedding from the foundation model.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            The input tensor to the model.
+
+        Returns
+        -------
+        The patch token embedding.
+        """
+        return self._model(x)[FMEmbedType.PATCH_TOKEN.value]
+
+    def _concat_embedding(self, x: torch.Tensor) -> Any:
+        """
+        Extract the concatenated class and patch token embeddings from the foundation model.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            The input tensor to the model.
+
+        Returns
+        -------
+        The concatenated class and patch token embeddings.
+        """
+        output = self._model(x)
+        cls_token = output[FMEmbedType.CLS_TOKEN.value].unsqueeze(1)
+        mean_patch_token = output[FMEmbedType.PATCH_TOKEN.value].mean(dim=1).unsqueeze(1)
+        tokens = torch.cat([cls_token, mean_patch_token], dim=1)
+        output = tokens.mean(dim=1)
+        return output
+
+    def _general_embedding(self, x: torch.Tensor) -> Any:
+        """
+        This function is for the foundation models that do not have specific embedding outputs.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            The input tensor to the model.
+
+        Returns
+        -------
+        The output of the model.
+        """
+        return self._model(x)
+
+    def forward(self, x: torch.Tensor) -> Any:
+        return self._forward_function(x)
+
+
+class SegmentationModel(AhcoreJitModel):
```

Comment:
Is there any specific logic for segmentation models in this class? It seems generic behavior, applicable for any jit model?